packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -36,7 +36,7 @@ import type {Widget} from "../../renderer";
 import "mafs/core.css";
 import "./mafs-styles.css";
 
-export type Props = {
+export interface StatefulMafsGraphProps {
```

Comment:
I noticed that in Mafs code it uses `interface` for Props types. In Perseus we've typically used `type`.  Is there a benefit/improvement from using `interface`? I don't have strong opinions other than a desire to be consistent. :)

Reply:
Sorry, this change came from [another PR](https://github.com/Khan/perseus/pull/1266) that this one was stacked on top of. https://github.com/Khan/perseus/pull/1266

The difference between `type` and `interface` is something I need to study more. The major difference I can think of is that only `interface`s can be `extend`ed. So you can do:

```ts
interface DerivedType extends Supertype {
  someProperty: string
}
```

whereas with `type`, your only option is

```
type DerivedType = Supertype & {someProperty: string}
```

which has surprising behavior when Supertype has a `someProperty` that's not a string. It looks like this may have been fixed in the latest version of TS to be less surprising (someProperty is correctly typed as the intersection of the property types from `Supertype` and `{someProperty: string}`) but IMO the `extends` behavior (subtype overrides properties of supertype) is more intuitive and useful.

To be super hand-wavy: OO subtyping and algebraic type intersection have different semantics and different use cases.

There is also a difference in editor tooling: JetBrains displays these little icons in the gutter to tell you which member of an interface a value is implementing.

<img width="481" alt="Screen Shot 2024-05-15 at 9 40 35 AM" src="https://github.com/Khan/perseus/assets/693920/dd8b86c7-1d21-46eb-a70b-cdbfcf01221b">

In this specific case, I don't think there's a practical difference between type and interface. I'll probably switch this back to `type` (on the other branch).